google/cloud/security/scanner/audit/org_rules_engine.py

```diff
@@ -58,17 +58,17 @@ def find_policy_violations(self, resource, policy, force_rebuild=False):
                 the rules definition file and add the rules to the book.
 
         Returns:
-            A list of tuples containing the policy diffs.
+            A generator of rule violations.
         """
         if self.rule_book is None or force_rebuild:
             self.build_rule_book()
 
         violations = []
-
+        # pylint: disable=redefined-variable-type
         for binding in policy.get('bindings', []):
-            binding_violations = (
+            violations = itertools.chain(
+                violations,
                 self.rule_book.find_violations(resource, binding))
-            violations.extend(binding_violations)
 
         return violations
```

Comment:
Since it is a small number, and maybe used or transformed by the caller, I would return a list or set object. Probably a list.